src/Illuminate/Support/Uri.php

```diff
@@ -0,0 +1,338 @@
+<?php
+
+namespace Illuminate\Support;
+
+use Closure;
+use Illuminate\Contracts\Routing\UrlRoutable;
+use Illuminate\Contracts\Support\Htmlable;
+use Illuminate\Contracts\Support\Responsable;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Traits\Conditionable;
+use Illuminate\Support\Traits\Tappable;
+use League\Uri\Contracts\UriInterface;
+use League\Uri\Uri as LeagueUri;
+use SensitiveParameter;
+use Stringable;
+
+class Uri implements Htmlable, Responsable
+{
+    use Conditionable, Tappable;
+
+    /**
+     * The URL generator resolver.
+     */
+    protected static ?Closure $urlGeneratorResolver = null;
```

Comment:
```suggestion
    /**
     * The URL instance.
     */
    protected UriInterface $uri;

    /**
     * The URL generator resolver.
     */
    protected static ?Closure $urlGeneratorResolver = null;
```

Fixes following deprecation:

```
1) /home/runner/work/framework/framework/src/Illuminate/Support/Uri.php:31
Creation of dynamic property Illuminate\Support\Uri::$uri is deprecated
```